src/main/java/geocode/kdtree/KDTree.java

```diff
@@ -44,7 +44,22 @@ public KDTree( List<T> items ) {
     }
 
     public T findNearest( T search ) {
-        return findNearest(root, search, 0).location;
+        return findNearest(search, null);
+    }
+
+    /**
+     * Finds nearest point within a max distance (in km)
+     */
+    public T findNearest(T search, Double maxDistance) {
+        T nearest = findNearest(root, search, 0).location;
+        if (null != maxDistance) {
+            int earthRadiusInKm = 6371;
+            double distanceInKm = Math.sqrt(nearest.squaredDistance(search)) * earthRadiusInKm;
```

Comment:
This _looks_ really confusing without any references to normalization... feels like we might need some extra documentation throughout this to keep it readable and maintainable in the long run.

Reply:
I'll add more documentation.